cleanlab/dataset.py

```diff
@@ -0,0 +1,416 @@
+    dataset. So a score of 1 suggests your data is perfectly labeled and a score of 0.5 suggests
+    that, on average across all classes, about half of the label may have issues. Thus, a higher
+    score implies higher quality labels, with 1 implying labels that have no issues.
+
+    Parameters
+    ----------
+    For parameter info, see the docstring of `dataset.find_overlapping_classes`
+
+    Returns
+    -------
+    health_score : float
+        A score between 0 and 1 where 1 implies the dataset has all estimated perfect labels.
+        A score of 0.5 implies that, on average, half of the dataset's label have estimated issues.
+    """
+
+    if joint is None:
+        joint = estimate_joint(
+            labels=labels,
+            pred_probs=pred_probs,
+            confident_joint=confident_joint,
+            multi_label=multi_label,
+        )
+    if num_examples is None:
+        num_examples = _get_num_examples(labels=labels)
+    joint_trace = joint.trace()
+    if verbose:
+        num_issues = (num_examples * (1 - joint_trace)).round().astype(int)
+        print(
+            f" * Overall, about {1 - joint_trace:.0%} ({num_issues:,} of the {num_examples:,}) "
+            f"labels in your dataset have issues.\n"
+            f" ** The overall label health score for this dataset is: {joint_trace:.2f}."
+        )
+    return joint_trace
+
+
+def health_summary(
+    labels=None,
+    pred_probs=None,
+    *,
+    asymmetric=False,
+    class_names=None,
+    num_examples=None,
+    joint=None,
+    confident_joint=None,
+    multi_label=False,
+):
+    """Prints a healthy summary of your datasets including results for powerful statistics like:
+    * the classes with the most and least label issues
+    * classes that overlap and could potentially be merged
+    * overall data label quality health score statistics for your dataset
+
+    Parameters
+    ----------
+    For parameter info, see the docstring of `dataset.find_overlapping_classes`
+
+    Returns
+    -------
+    health_score : float
+        A score between 0 and 1 where 1 implies the dataset has all estimated perfect labels.
+        A score of 0.5 implies that, on average, half of the dataset's label have estimated issues.
+    """
+
+    if joint is None:
+        joint = estimate_joint(
+            labels=labels,
+            pred_probs=pred_probs,
+            confident_joint=confident_joint,
+            multi_label=multi_label,
+        )
+    if num_examples is None:
+        num_examples = _get_num_examples(labels=labels)
+
+    print(
+        "*" * 56
+        + "\n"
+        + "Generating a Cleanlab Dataset Health Summary\n"
+        + f" for your dataset with {num_examples:,} examples and {len(joint):,} classes.\n"
+        + "Note, Cleanlab is not a medical doctor... yet.\n"
+        + "*" * 56
+        + "\n",
+    )
+
+    df_class_label_quality = rank_classes_by_label_quality(
```

Comment:
Some of the DF column names could be more clear if using terms from our Studio report-template instead.

Eg:

"Overall Label Issues" -> "Num. Label Issue"

Reply:
Could add informative column from our Studio report template:
"% of Examples with Label Issue" 
which is supposed to = 100 * (num-examples with this given label and flagged as having issue) / (num-examples with this given label)

Reply:
Also could add informative column from our Studio report template:
"Number of examples given this label"
which helps user trade-off between prioritizing classes that are very noisy but rare overall vs classes that are somewhat noisy but very common overall.